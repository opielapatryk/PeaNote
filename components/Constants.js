export const userLink = (user)=>{
    return `http://localhost:8000/api/users/${user}/`
}

export const usersLink = ()=>{
    return `http://localhost:8000/api/users/`
}

export const stickerLink = (sticker)=>{
    return `http://localhost:8000/api/stickers/${sticker}/`
}

export const stickersLink = ()=>{
    return `http://localhost:8000/api/stickers/`
}