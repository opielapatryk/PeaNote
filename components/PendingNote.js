import { Text, View, Pressable } from 'react-native';
import React from 'react';
import { styles } from '../assets/styles/styles';
import { useDispatch, useSelector} from 'react-redux';
import { handlePress } from '../logic/funcPendingNote';

export function PendingNote({ id, text, isInfo }) {
  const {pendingNotes} = useSelector((state) => state.board);
  const dispatch = useDispatch();

  return (
    <Pressable style={styles.note} onPress={()=>handlePress(pendingNotes,dispatch,isInfo,id)}>
      <View>
        <Text>{pendingNotes.find((item) => item.id === id)?.text}</Text>
      </View>
    </Pressable>
  );
}