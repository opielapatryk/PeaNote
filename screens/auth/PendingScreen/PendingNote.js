import { Text, View, Pressable } from 'react-native';
import React from 'react';
import { styles } from '../../../assets/styles/styles';
import { useDispatch, useSelector} from 'react-redux';
import { handlePress } from './funcPendingNote';

export function PendingNote({ id, isInfo }) {
  const {pendingNotes} = useSelector((state) => state.board);
  const dispatch = useDispatch();

  return (
    <Pressable style={styles.note} onPress={()=>handlePress(pendingNotes,dispatch,isInfo,id)}>
        <Text style={styles.noteText}>{pendingNotes.find((item) => item.id === id)?.text}</Text>
    </Pressable>
  );
}