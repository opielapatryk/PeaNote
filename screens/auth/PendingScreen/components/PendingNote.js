import { Text, Pressable,View,Dimensions} from 'react-native';
import React from 'react';
import { styles } from '../../../../assets/styles/styles'
import { useDispatch, useSelector} from 'react-redux';
import { handlePress } from '../functions/funcPendingNote';
import { sendNoteToBoard } from '../functions/sendNoteToBoard';
import { deleteNote } from '../functions/deleteNote';
import { removePendingNote ,addNote} from '../../../../store/notes/boardSlice';

export function PendingNote({ id, isInfo, content,creator }) {
  const {pendingNotes} = useSelector((state) => state.board);
  const dispatch = useDispatch();

  return (
    <Pressable onPress={()=>handlePress(pendingNotes,dispatch,isInfo,id)}>
    {!isInfo&&<>
      <View style={styles.note}>
      <Text style={styles.noteTextHeader}>{pendingNotes.find((item) => item.id === id)?.creator}</Text>
        <Text style={styles.noteText}>{pendingNotes.find((item) => item.id === id)?.text}</Text>
      </View>

      </>}

      {isInfo&&<>
      <View style={styles.noteclicked}>
      <Pressable style={{height:Dimensions.get("window").height / 5,width:Dimensions.get('window').width / 4.4,justifyContent:'center',paddingLeft:Dimensions.get('window').width / 20}} onPress={() => {
              deleteNote(id);
              dispatch(removePendingNote(id));
            }}>
        <Text style={{fontWeight:'bold',fontSize:20}}>Click{"\n"}here{"\n"}to{"\n"}remove</Text>
      </Pressable>
<View style={{height:Dimensions.get("window").height / 8,borderLeftWidth:1,alignSelf:'center'}}></View>
        <Pressable style={{height:Dimensions.get("window").height / 5,width:Dimensions.get('window').width / 4.4,justifyContent:'center',paddingLeft:Dimensions.get('window').width / 25}} onPress={() => {
              sendNoteToBoard(id);
              dispatch(addNote({ id: id, text: content, isInfo: false, creator:creator}));
              dispatch(removePendingNote(id));
            }}>
          <Text style={{fontWeight:'bold',fontSize:20}}>Click{"\n"}here{"\n"}to{"\n"}approve</Text>
        </Pressable>
      </View>
      
      </>}
    </Pressable>
  );
}
