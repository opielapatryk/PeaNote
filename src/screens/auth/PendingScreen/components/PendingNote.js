import { Text, Pressable,View,ActivityIndicator} from 'react-native';
import React, { useState } from 'react';
import { styles } from '../../../../../assets/styles/styles'
import { useDispatch, useSelector} from 'react-redux';
import { handlePress } from '../functions/funcPendingNote';
import { sendNoteToBoard } from '../functions/sendNoteToBoard';
import { deleteNote } from '../functions/deleteNote';
import { removePendingNote} from '../../../../store/notes/boardSlice';
import firestore from '@react-native-firebase/firestore'
import auth, {firebase} from '@react-native-firebase/auth'

export function PendingNote({ id, isInfo, content,creator }) {
  const {pendingNotes} = useSelector((state) => state.board);
  const dispatch = useDispatch();
  const [isLoadingLeft,setLoadingLeft] = useState(false)
  const [isLoadingRight,setLoadingRight] = useState(false)
  const [nickname, setNickname] = useState()

  const getNickname = async () => {
    const EMAIL = auth().currentUser.email
    const currentUser = await firestore().collection('users').where('email', '==', EMAIL).get()
  
    if (currentUser.docs && currentUser.docs.length > 0) {
      const friend = currentUser.docs[0].data().friends.find(
        (friend) => friend.username === creator || friend.email === creator
      );
  
      if (friend === undefined) {
        return creator;
      }
  
      return friend.nickname;
    } else {
      return creator; // Handle the case where docs array is undefined or empty
    }
  }
  
    getNickname().then((nickname)=>{
      setNickname(nickname)
    })

  return (
    <Pressable onPress={()=>handlePress(pendingNotes,dispatch,isInfo,id)}>
    {!isInfo&&<>
      <View style={styles.note}>
      <Text style={styles.noteTextHeader}>
        {nickname?nickname:pendingNotes.find((item) => item.id === id)?.creator}
        </Text>
        <Text style={styles.noteText}>{pendingNotes.find((item) => item.id === id)?.text}</Text>
      </View>

      </>}

      {isInfo&&<>
      <View style={styles.noteclicked}>
        <Pressable style={styles.noteIsInfoTrueLeftButton} onPress={ async () => {
                setLoadingLeft(true)
                await deleteNote(content,creator);
                dispatch(removePendingNote(id));
                setLoadingLeft(false)
              }}>
          {isLoadingLeft?<ActivityIndicator size={'large'} color={'black'}/>:<Text style={styles.noteIsInfoTrueButtonsText}>Click{"\n"}here{"\n"}to{"\n"}remove</Text>}
        </Pressable>

        <View style={styles.notesIsInfoVerticalLine}></View>

        <Pressable style={styles.noteIsInfoTrueRightButton} onPress={async () => {
              setLoadingRight(true);
              await sendNoteToBoard(content,creator);
              dispatch(removePendingNote(id));
              setLoadingRight(false)
            }}>
              {isLoadingRight?<ActivityIndicator size={'large'} color={'black'}/>:<Text style={styles.noteIsInfoTrueButtonsText}>Click{"\n"}here{"\n"}to{"\n"}approve</Text>}
          
        </Pressable>
      </View>
      
      </>}
    </Pressable>
  );
}
