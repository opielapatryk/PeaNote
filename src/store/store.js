import { configureStore } from '@reduxjs/toolkit'
import boardReducer from './notes/boardSlice'
import settingsReducer from './settings/settingsSlice'

export default configureStore({
    reducer:{
        board:boardReducer,
        settings:settingsReducer
    }
})