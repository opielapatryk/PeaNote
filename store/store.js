import { configureStore } from '@reduxjs/toolkit'
import boardReducer from './notes/boardSlice'

export default configureStore({
    reducer:{
        board:boardReducer
    }
})