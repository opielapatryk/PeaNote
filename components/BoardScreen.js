import React, {useState} from 'react';
import {ScrollView,Pressable,Button,TextInput,View,FlatList,Text} from 'react-native';
import {Note} from './Note'
import { useDispatch, useSelector} from 'react-redux';
import {styles} from '../assets/styles';
import {removeNote,addNote,showInput,hideInput} from '../store/boardSlice';
// import { setText,isNoteTrue,hideInfo ,showInfo } from '../store/noteSlice';

const BoardScreen = () => {
    const { notes,isInput } = useSelector((state)=>state.board)
    // const { id, text, isNote, isInfo } = useSelector((state) => state.note);
    const dispatch = useDispatch()
    const [text, setText] = useState('')
    const [id, setId] = useState(0)
    const [isNote, setIsNote] = useState(true)
    const [info, setIsInfo] = useState(false)

    if(isInput){
        return(
        <View>
            <TextInput placeholder='Insert text and press button to create note' onChangeText={(text)=>setText(text)}>

            </TextInput>

            <Button title='create note' onPress={()=>{
                dispatch(addNote({ id: notes.length + 1,text: text, isNote:true, isInfo:false }))
                // dispatch(setText(''));
                dispatch(hideInput());
            }}/>

            <Pressable onPress={()=>{
            //  dispatch(hideInfo())
            setIsInfo(false)
             if(isInput){
                 dispatch(hideInput())
             }else{
                 dispatch(showInput())
             }
            }} style={styles.board}>

                <ScrollView>
                    {notes.map((note) => (
                        <Pressable key={note.id} onPress={()=>{
                            if (info) {
                                // dispatch(removeNote())
                                // filter notes item id !== id 
                              } else {
                                // dispatch(showInfo())
                                setIsInfo(true)
                              }
                        }}><Note key={note.id} id={note.id} text={note.text} isNote={note.isNote} isInfo={note.isInfo} /></Pressable>
                    ))}
                </ScrollView>
            </Pressable>
        </View>
    )}

    return (
        <Pressable onPress={()=>{
            // dispatch(isNoteTrue())
            console.log(notes);
            // dispatch(hideInfo())
            setIsInfo(false)
            if(isInput){
                dispatch(hideInput())
            }else{
                dispatch(showInput())
            }
        }} style={styles.board}>
            <FlatList
                keyExtractor={(item) => item.id.toString()}
                data={notes}
                renderItem={({ item }) => <Pressable key={item.id} onPress={()=>{
                    if (info) {
                        dispatch(removeNote())
                      } else {
                        // dispatch(showInfo())
                        setIsInfo(true)
                      }
                }}><Note key={item.id} id={item.id} text={item.text} isNote={item.isNote} isInfo={item.isInfo} /></Pressable>}
            />
        </Pressable>
    );
}


export default BoardScreen;