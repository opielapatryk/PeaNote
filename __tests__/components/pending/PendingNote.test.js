import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';  
import store from '../../../src/store/store'; 
import {PendingNote} from '../../../src/screens/auth/PendingScreen/components/PendingNote';

jest.mock('@react-native-firebase/auth', () => () => {
    return {
        currentUser: {
            email:'test@example.com'
        }
    }
  })

  jest.mock('@react-native-firebase/firestore', () =>()=> {
    return {
      collection: jest.fn(()=>{
        return{
          where: jest.fn(()=>{
            return{
              get: jest.fn(() => Promise.resolve(true))
            }
          }),
        }
      }),
    };
  })


test('Pending note renders correctly.', async () => {
  const tree = renderer.create(
    <Provider store={store}>
        <PendingNote id={1} isInfo={false}/>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
