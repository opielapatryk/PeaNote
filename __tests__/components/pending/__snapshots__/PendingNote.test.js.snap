// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pending note renders correctly. 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      {
        "backgroundColor": "#f9e889",
        "borderRadius": 5,
        "height": 266.8,
        "margin": 8,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
        "shadowOffset": {
          "height": 1,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 2,
        "width": 340.9090909090909,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "600",
          "marginBottom": 5,
        }
      }
    />
    <Text
      style={
        {
          "fontSize": 15,
        }
      }
    />
  </View>
</View>
`;
